lib/nerves_runtime/heart.ex

```diff
@@ -0,0 +1,131 @@
+defmodule Nerves.Runtime.Heart do
+  @moduledoc """
+  Functions for querying Nerves Heart and the device's watchdog
+
+  Nerves Heart integrates Erlang's
+  [heart](https://www.erlang.org/doc/man/heart.html) process with a hardware
+  watchdog.  This makes it possible for a device to recover from a hang. The
+  way it works is that the Erlang runtime regularly checks that it's ok. If so,
+  it sends a message to `heart`. Nerves heart then pets the hardware watchdog.
+  If messages ever stop being sent to `heart`, the hardware watchdog will trip
+  and reboot the device. You can add additional health checks for your
+  application by providing a callback to `:heart.set_callback/2`.
+  """
+
+  @typedoc """
+  Nerves Heart's current status
+
+  * `:program_name` - `"nerves_heart"`
+  * `:program_version` - Nerves heart's version number
+  * `:identity` - The hardware watchdog that's being used
+  * `:firmware_version` - An integer that represents the hardware watchdog's firmware revision
+  * `:options` - Hardware watchdog options as reported by Linux
+  * `:time_left` - How many seconds are left before the hardware watchdog triggers a reboot
+  * `:pre_timeout` - How many seconds before the watchdog expires that Linux
+    will receive a pre-timeout notification * `:timeout` - The hardware watchdog
+    timeout. This is only changeable in the Linux configuration
+  * `:last_boot` - What caused the most recent boot. Whether this is reliable
+    depends on the watchdog.
+  * `:heartbeat_timeout` - Erlang's heartbeat timeout setting. Note that the
+    hardware watchdog timeout supersedes this since it reboots.
+  """
+  @type info() :: %{
+          program_name: String.t(),
+          program_version: Version.t(),
+          identity: String.t(),
+          firmware_version: non_neg_integer(),
+          options: non_neg_integer() | [atom()],
+          time_left: non_neg_integer(),
+          pre_timeout: non_neg_integer(),
+          timeout: non_neg_integer(),
+          last_boot: :power_on | :watchdog,
+          heartbeat_timeout: non_neg_integer()
+        }
+
+  @doc """
+  Return whether Nerves heart is running
+
+  If you're using a Nerves device, this always returns `true` except possibly
+  when porting Nerves to new hardware. It is a quick sanity check.
+  """
+  def running_nerves_heart?() do
```

Comment:
```suggestion
  @spec running?() :: boolean()
  def running?() do
```

No specs?
Also, just a suggestion to reduce redundancy (which may just be a Jon thing)